src/esp/assets/ResourceManager.cpp

```diff
@@ -1985,6 +1989,112 @@ void ResourceManager::removePrimitiveMesh(int primitiveID) {
   primitive_meshes_.erase(primitiveID);
 }
 
+bool ResourceManager::importAsset(const std::string& filename,
+                                  bool requiresLighting) {
+  bool meshSuccess = true;
+  if (resourceDict_.count(filename) > 0) {
+    return true;
+  } else {
+    esp::assets::AssetInfo meshinfo{AssetType::UNKNOWN, filename};
+    meshinfo.requiresLighting = requiresLighting;
+    meshSuccess = loadRenderAssetGeneral(meshinfo);
+
+    // check if collision handle exists in collision mesh groups yet.  if not
+    // then instance
+    if (collisionMeshGroups_.count(filename) == 0) {
+      // set collision mesh data
+      const MeshMetaData& meshMetaData = getMeshMetaData(filename);
+
+      int start = meshMetaData.meshIndex.first;
+      int end = meshMetaData.meshIndex.second;
+      //! Gather mesh components for meshGroup data
+      std::vector<CollisionMeshData> meshGroup;
+      for (int mesh_i = start; mesh_i <= end; ++mesh_i) {
+        GenericMeshData& gltfMeshData =
+            dynamic_cast<GenericMeshData&>(*meshes_[mesh_i].get());
+        CollisionMeshData& meshData = gltfMeshData.getCollisionMeshData();
+        meshGroup.push_back(meshData);
+      }
+      collisionMeshGroups_.emplace(filename, meshGroup);
+    }
+  }
+
+  return meshSuccess;
+}
+
+std::string ResourceManager::setupMaterialModifiedAsset(
+    const std::string& filename,
+    const Magnum::Color4& matColor,
+    const Magnum::Color3& specularColor) {
+  assert(resourceDict_.count(filename) > 0);
+
+  std::string modifiedAssetName = filename + "_MatMod";
```

Comment:
Another good point. 

Currently, this is setup to support only the URDF case where every time we load an asset we know the "override" material. We adjust the MeshMetaData material keys and then immediately create a Drawable. The Drawable never references back to the resourceDict_ which only stores metadata, so it doesn't matter if the entries are changed after that point to another material override. 

We also key the resourceDict_ by contrived keys in other cases (E.g. primitives, convex decomps, voxel meshes). The filepath keys should end with a file ending (e.g. .glb), so "my_file.glb_MatMod" should not be possible to load from disk.

Reply:
@aclegg3 Can we just store a filesystem invalid seperator? Like ?, *, and "

Reply:
Sure, easy enough. @jturner65 has been using this method in other places also I think.

Reply:
Yep. See [this link](https://github.com/facebookresearch/habitat-sim/blob/9bd03ecefde715de505a193ab454c1fb380a477a/src/esp/core/managedContainers/ManagedContainerBase.cpp#L135)

Reply:
@aclegg3 and I discussed offline and we had an idea to embed material-override info in our AssetInfo struct and conceptually handle this override at asset-load time, instead of later. This addresses/sidesteps my long-standing concern about modifying assets after loading them (and how to propagate those changes).